src/Http/Middleware/AddIdpUserDataMiddleware.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Zanichelli\IdpExtension\Http\Middleware;
+
+use Closure;
+use Exception;
+use GuzzleHttp\Client;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Zanichelli\IdpExtension\Models\ZTrait\ZUserBuilder;
+
+class AddIdpUserDataMiddleware
+{
+    use ZUserBuilder;
+
+    public function handle(Request $request, Closure $next, string $idpUrlEnvKey = 'IDP_URL')
+    {
+        $token = $request->cookies->get(config("idp.cookie.name"));
+        if ($token) {
+            try {
+                $client = new Client(['verify' => false]);
+                $res = $client->get(env($idpUrlEnvKey) . '/v1/user?token=' . $token);
```

Comment:
Qui si può usare `env('IDP_TOKEN_URL')`, come nell'altro middleware, senza bisogno di passare in ingresso `$idpUrlEnvKey`